### PR TITLE
fix: allow the modal process to finish even when home changes

### PR DESCRIPTION
**Pull request by hansl**
The bug happened because the account was selected, causing the Home page
to transfer to listing the account assets, causing the Modal component to
be garbage collected. Instead of waiting for the modal to be completed
before choosing the account (which would have moved a lot of logic
outside the modal), moving the modal ownership to the top-level app
was the least amount of change to fix the bug.

This code needs a good overhaul at some point. But this wont be the time
nor the place.

Fixes https://github.com/liftedinit/alberto/issues/111

### CODE REVIEW

**Comment by stanleyjones** [comment]
Is this ready for review? I see some console logging still in there.

File: src/features/accounts/components/accounts-menu/add-account-modal.tsx [hunk 1]
```diff
@@ -54,15 +54,18 @@ export function AddAccountModal({
   React.useEffect(() => {
     setAddMethod("")
   }, [isOpen])
+  React.useEffect(() => {
+    console.log(addMethod, hasAddMethod);
+  }, [addMethod, hasAddMethod])
```
Comment (stanleyjones) [inline src/features/accounts/components/accounts-menu/add-account-modal.tsx:59]: Is this still needed?

File: src/views/app/app.tsx [hunk 2]
```diff
@@ -21,29 +23,41 @@ export function App() {
...
-        <Route path="contacts" element={<Contacts />} />
-        <Route path="send" element={<SendAsset />} />
-      </Route>
-    </Routes>
-  )
+      </Routes>
+      <AddAccountModal
+          isOpen={isAddAccountOpen}
+          onClose={onCloseAddAccount}
+      />
```
Comment (stanleyjones) [inline src/views/app/app.tsx:60]: oh jeez 😮 

I mean, I guess I asked for this when I said it was fine to do the simplest thing that could work.

**Review by stanleyjones — COMMENTED** [review]
I'm not convinced this won't introduce other errors with other modals. I'm going to play with it manually a bit.

File: src/views/app/app.tsx (hunk 2 above)
Comment (hansl) [reply src/views/app/app.tsx:60]: If you need details I'm free to discuss.

**Review by stanleyjones — APPROVED** [review]
Looks good.

**Pull request merged**